app/controllers/room/hero_images_controller.rb

```diff
@@ -0,0 +1,26 @@
+class Room
+  class HeroImagesController < ApplicationController
+    def create
+      old_hero_image = room.hero_image
+      new_media = Media.create
+      new_media.upload.attach(media_params[:upload])
+
+      if room.update(hero_image: new_media)
+        old_hero_image&.destroy
```

Comment:
I wonder if there's a way to push responsibility for deleting the old hero image into the model...

If we overload `Room#hero_image=`, that might work... but I actually don't know and don't know that it's worth diving into unless it feels like a solid learning experience.